In [11]:
# Import the necessary classes
import numpy as np
from sklearn import datasets

# Load and parse the data file
iris = datasets.load_iris()
iris_X = iris.data
iris_Y = iris.target
np.unique(iris_Y)

# Split iris data in train and test data
# A random permutation, to split the data randomly
np.random.seed ( 0 )
indices = np.random.permutation(len(iris_X))
# Take some elements from the shuffled array
iris_X_train = iris_X[indices[:-10]]
iris_Y_train = iris_Y[indices[:-10]]
iris_X_test = iris_X[indices[-10:]]
iris_Y_test = iris_Y[indices[-10:]]

## Question 1 - Compute Error.
E = 1 - accuracy
WIth the default k value of 5, the error is:

In [83]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier()
knn.fit(iris_X_train, iris_Y_train)

from sklearn.metrics import accuracy_score
iris_prediction = knn.predict(iris_X_train)
iris_prediction
iris_Y_test
accuracy_score(iris_Y_test, knn.predict(iris_X_test))

0.90000000000000002

## Question 2 - Accuracy scores with different k
A few different values for k have been tried.

| K Value   | Accuracy |
|:---------:|:----------:|
|1->7|0.90000000000000002|
|8->12|1.0|
|13->15|0.90000000000000002|
|16->92|1.0|
|93->136|0.80000000000000004|
|137-138|0.59999999999999998|
|139-140|0.20000000000000001|
